content/doc/book/system-administration/reverse-proxy-configuration-with-jenkins.adoc

```diff
@@ -0,0 +1,1389 @@
+_Enable proxy compatibility_ if you have already enabled _Prevent Cross
+Site Request Forgery exploits_ 
+. Go to [.nolink]#https://jenkins.acme.example/manage#
+. You will still experience the "It appears that your reverse
+proxy set up is broken." as expected
+.. If you do not get that at this point, then that is very weird...
+Continue anyway.
+. Right click the _Configure System_ link and choose to
+inspect the element.
+.. Make sure you are still on the Manage page as you will want
+it as your referrer
+. Change the value of the _href_ attribute to be
+_administrativeMonitor/hudson.diagnosis.ReverseProxySetupMonitor/test_
+. Open the link you just changed in a new tab.
+.. Keep this tab open
+. Observe the "[.nolink]#https://jenkins.acme.example/manage#
+vs http:" error and bask in its glory
+.. a white page served with HTTP status code is 200 indicates
+all is well
+... If you do get that at this point, then that is very
+weird... Continue anyway.
+
+===== [.nolink]#Fixing the errors#
+
+. In IIS Manager got to _Application Pools_ then edit
+_DefaultAppPool_ so that the _.NET CLR version_ is *No Managed Code* 
+.. You might find that this is not necessary (at far as you
+can tell) for your setup, since IIS will only act as a TLS/SSL
+offloading reverse proxy, we don't need it. IIS on Windows 7 does appear
+to need this to be turned off for some urls
+(https://jenkins.acme.example/*)
+. Then go to _Sites_ → _Default Web Site_ → _Request
+Filtering_ and in the _Actions_ panel choose _Edit Feature Settings..._
+and turn on *Allow doube escaping*
+.. This is so IIS forwards URLs like
+[.nolink]#https://jenkins.acme.example/%2525# to Jenkins instead of
+showing an IIS error page
+. Last, but not least, go to _Sites_ → _Default Web
+Site_ → __Configuration Editor__ and change the _Section_ to
+_system.webServer/rewrite/rules_
+. Now you should see the URL Rewrite 2.1 property
+_useOriginalURLEncoding_ listed, if not install URL Rewrite 2.1 using
+the x86 or x64 installer, not the WebPI one and resume from here after a
+reboot.
+. Change _useOriginalURLEncoding_ to *False*
+.. As the URL Rewrite 2.1 announcement this will change the
+value of \{UNENCODED_URL} to make it _RFC3986_ and usable for reverse
+proxy forwarding purposes
+.. original as in pre 2.1 behaviour. 
+. Refresh that tab you were supposed to keep open, or recreate
+it.
+.. Again, take some time to bask in its glory 
+. It should now be white, also the Manage page should no
+longer complain!
+
+===== Continue configuring IIS
+
+Some of the things you might want but I won't cover:
+
+* _Hypertext Strict Transport Security_ headers
+* Redirecting from non canonical URLs to the canonical URL
+(ok, sort of covered this in the web.config example) 
+* The X-UA-Compatibility header so that Internet Explorer 11
+(or 9, or ...) won't claim to be IE 7 for intranet sites
+* Use IIS Crypto to configure cipher suites
+* ...
+
+===== A working web.config
+
+*web.config*
+
+[source,xml]
+----
```

Comment:
Could this example have its indentation switched from 4 spaces to 2 spaces and have the longer lines split in a way that is still consistent with valid XML data?

It creates such long examples that the page is unreadable on all but my widest screens